gtk_v4l_device_supports_ctrl_events: Return true iff channel is NULL

Previously, the function could return false if the channel pointer is
non-NULL, but happened to be aligned to a 4 GiB boundary.

GtkComboBoxText functions need casts

Otherwise, current compilers error out because GtkComboBox *
and GtkComboBoxText * are distinct types that cannot be converted
implicitly.

Submitted upstream: <https://github.com/jwrdegoede/gtk-v4l/pull/2>

diff --git a/lib/gtk-v4l-device.c b/lib/gtk-v4l-device.c
index b982b19a4300cf4e..f87d40d670bb774a 100644
--- a/lib/gtk-v4l-device.c
+++ b/lib/gtk-v4l-device.c
@@ -401,5 +401,5 @@ void gtk_v4l_device_update_controls (Gtkv4lDevice *self)
 
 gboolean gtk_v4l_device_supports_ctrl_events (Gtkv4lDevice *self)
 {
-  return self->priv->channel;
+  return self->priv->channel != NULL;
 }
diff --git a/src/gtk-v4l.c b/src/gtk-v4l.c
index 0eddf3e651b4389c..b37b89ee2b049683 100644
--- a/src/gtk-v4l.c
+++ b/src/gtk-v4l.c
@@ -68,7 +68,7 @@ v4l2_combo_add_device(Gtkv4lDeviceList *devlist,
   gint active;
   GtkComboBox *combo = GTK_COMBO_BOX(user_data);
 
-  gtk_combo_box_text_insert_text (combo, idx, device->card);
+  gtk_combo_box_text_insert_text (GTK_COMBO_BOX_TEXT (combo), idx, device->card);
   active = gtk_combo_box_get_active (combo);
   if (active == -1)
   {
@@ -88,7 +88,7 @@ v4l2_combo_remove_device(Gtkv4lDeviceList *devlist,
   /* If this removes the current device
      v4l2_combo_change_device_cb() will get called and that will
      handle selecting a new device. */
-  gtk_combo_box_text_remove (combo, idx);
+  gtk_combo_box_text_remove (GTK_COMBO_BOX_TEXT (combo), idx);
   if ((gtk_tree_model_iter_n_children( gtk_combo_box_get_model (combo),NULL)) == 0)
     gtk_widget_set_sensitive (default_button, FALSE);
 }
